umbraco/Umbraco-CMS
# Default value editors for all (meaningful) value types

### Prerequisites


### Description

This PR adds default value editors with no attached configuration for all meaningful value types:

- TEXT & STRING (`Umbraco.Plain.String`)
- JSON (`Umbraco.Plain.Json`)
- DECIMAL (`Umbraco.Plain.Decimal`)
- DATE & DATETIME (`Umbraco.Plain.DateTime`)
- INTEGER (`Umbraco.Plain.Integer`)
- TIME (`Umbraco.Plain.Time`)

Value types XML and BIGINT has been left out on purpose. If demand arises, we can add them later on.

Notice that `Umbraco.Plain.String` actually stores property data as TEXT. This was decided internally to improve DX. Alternatively we would need an extra `Umbraco.Plain.Text` value editor, which isn't the most friendly approach.

The aliases for these editors are suffixed with `Umbraco.Plain` (i.e. `Umbraco.Plain.String`) to indicate that these editors do *not* have any configuration nor out-of-the-box interpretation.

This PR also exposes the editor UI alias of datatypes, so custom property editors can use this when evaluating whether they can handle any given published property.

To be able to test all this, #14869 (default property value converters for all value types) has been cherry-picked from V13 into this PR.

### Testing this PR

1. Create property editors for the new backoffice using all of the new default editors as property editor schemas.
2. Create a content type with properties based on these property editors.
3. Create and save content with each of these properties populated.
4. Since we can't yet publish content in the new backoffice, use the Delivery API to fetch the content in preview mode.
5. Verify that the property outputs are mapped correctly to the expected types (i.e. decimal values are output as decimals, not strings).
6. Create a custom value converter that can convert the above created property editors based on their editor UI alias.
7. Verify that the property outputs (via the Delivery API) are now handled by this new value converter.

If you are not well versed in any of the above, get in touch with @kjac to do some peer testing 😆
